userspace/libsinsp/examples/test.cpp

```diff
@@ -38,254 +48,347 @@ static sinsp_evt_formatter* default_formatter = nullptr;
+				long_options, &long_index)) != -1)
+	{
+		switch(op)
+		{
+		case 'h':
+			usage();
+			exit(EXIT_SUCCESS);
+		case 'f':
+			filter_string = optarg;
+			break;
+		case 'j':
+			// Initialize JSON formatters
+			default_formatter = new sinsp_evt_formatter(&inspector, DEFAULT_OUTPUT_STR);
+			process_formatter = new sinsp_evt_formatter(&inspector, PROCESS_DEFAULTS);
+			net_formatter = new sinsp_evt_formatter(&inspector, PROCESS_DEFAULTS " %fd.name");
+
+			inspector.set_buffer_format(sinsp_evt::PF_JSON);
+			dump = json_dump;
+			break;
+		case 'a':
+			g_all_threads = true;
+			break;
+		case 'e':
+			engine_string = optarg;
+			break;
+		case 'b':
+			bpf_path = optarg;
+			break;
+		case 'd':
+			buffer_dim = strtoul(optarg, NULL, 10);
+			break;
+		case 's':
+			file_path = optarg;
+			break;
+		default:
+			break;
+		}
+	}
+}
+#endif /* WIN32 */
+
+void open_engine(sinsp& inspector)
+{
+	std::cout << "-- Try to open: '" + engine_string + "' engine." << std::endl;
+
+	if(!engine_string.compare(KMOD_ENGINE))
+	{
+		inspector.open_kmod(buffer_dim);
+	}
+	else if(!engine_string.compare(BPF_ENGINE))
+	{
+		if(bpf_path.empty())
+		{
+			std::cerr << "You must specify the path to the bpf probe if you use the 'bpf' engine" << std::endl;
+			exit(EXIT_FAILURE);
+		}
+		else
+		{
+			std::cerr << bpf_path << std::endl;
+		}
+		inspector.open_bpf(buffer_dim, bpf_path.c_str());
+	}
+	else if(!engine_string.compare(UDIG_ENGINE))
+	{
+		inspector.open_udig(buffer_dim);
+	}
+	else if(!engine_string.compare(NODRIVER_ENGINE))
+	{
+		inspector.open_nodriver();
+	}
+	else if(!engine_string.compare(SAVEFILE_ENGINE))
+	{
+		if(file_path.empty())
+		{
+			std::cerr << "You must specify the path to the file if you use the 'savefile' engine" << std::endl;
+			exit(EXIT_FAILURE);
+		}
+		inspector.open_savefile(file_path.c_str(), 0);
+	}
+	else if(!engine_string.compare(SOURCE_PLUGIN_ENGINE))
+	{
+		std::cerr << "Not supported right now." << std::endl;
+		exit(EXIT_SUCCESS);
+	}
+	else if(!engine_string.compare(GVISOR_ENGINE))
+	{
+		std::cerr << "Not supported right now." << std::endl;
+		exit(EXIT_SUCCESS);
+	}
+	else if(!engine_string.compare(MODERN_BPF_ENGINE))
+	{
+		inspector.open_modern_bpf(buffer_dim);
+	}
+	else if(!engine_string.compare(TEST_INPUT_ENGINE))
+	{
+		std::cerr << "Not supported right now." << std::endl;
+		exit(EXIT_SUCCESS);
+	}
+	else
+	{
+		std::cerr << "Unknown engine" << std::endl;
+		exit(EXIT_FAILURE);
+	}
+
+	std::cout << "-- Engine '" + engine_string + "' correctly opened." << std::endl;
 }
 
 //
 // Sample filters:
 //   "evt.category=process or evt.category=net"
 //   "evt.dir=< and (evt.category=net or (evt.type=execveat or evt.type=execve or evt.type=clone or evt.type=fork or evt.type=vfork))"
 //
-int main(int argc, char **argv)
+int main(int argc, char** argv)
 {
-    sinsp inspector;
-    string filter_string;
-    std::function<void (sinsp& inspector)> dump = plaintext_dump;
+	sinsp inspector;
+	dump = plaintext_dump;
```

Comment:
Given this is an example binary, I think we can remove all of the plaintext_dump support as we need to try to also remove lines of codes wherever we can. JSON combined with bash core utils should cover all use cases?